Donation/DonationManager.swift

```diff
@@ -0,0 +1,98 @@
+/*****************************************************************************
+ * DonationViewController.swift
+ * VLC for iOS
+ *****************************************************************************
+ * Copyright (c) 2018 VideoLAN. All rights reserved.
+ * $Id$
+ *
+ * Authors: Carola Nitz <nitz.carola # googlemail.com>
+ *
+ *
+ * Refer to the COPYING file of the official project for license.
+ *****************************************************************************/
+import PassKit
+protocol DonationManagerDelegate: PKPaymentAuthorizationViewControllerDelegate {
+    func donationManagerAmountForDonation(donationManager: DonationManager) -> NSNumber
+    func donationManagerShouldPresentViewController(donationManager: DonationManager, viewController: UIViewController)
+}
+
+class DonationManager: NSObject {
+
+    let passLibrary = PKPassLibrary()
+    weak var delegate: DonationManagerDelegate?
+
+    @objc func donateAmount() {
+        guard let amount = delegate?.donationManagerAmountForDonation(donationManager: self) else {
+            assertionFailure("no amount was defined")
+            return
+        }
+
+        let request = PKPaymentRequest()
+        guard let currencyCode = Locale.current.currencyCode, let countrycode = Locale.current.regionCode else {
+            return assertionFailure("missing currencycode or reegioncode")
+        }
+        request.currencyCode = currencyCode
+        request.countryCode = countrycode
+        request.merchantIdentifier = "merchant.com.example.vlc-ios"
```

Comment:
Should we automagically use the bundle ID here?

Reply:
can you tell me how? because I only found how to use it in info.plists